dhall/src/builtins.rs

```diff
@@ -401,6 +409,62 @@ fn apply_builtin(b: Builtin, args: Vec<Nir>, env: NzEnv) -> NirKind {
             }
             _ => Ret::DoneAsIs,
         },
+        (Builtin::TextReplace, [needle, replacement, haystack]) => {
+            match (&*needle.kind(), &*replacement.kind(), &*haystack.kind()) {
+                (TextLit(n_lit), TextLit(r_lit), TextLit(h_lit)) => {
```

Comment:
Might the replacement not be allowed to be completely abstract? Eg. `\(x : Text) -> Text/replace "a" x "aaaaaa"`

Reply:
Yes, that is supported!

Reply:
Matching `replacement` with `TextLit(..)` prevents it from being something like a variable though

Reply:
I think what you want it to replace the needle with `InterpolatedText::Expr(replacement)` or something like that. You only need to check if `replacement` is a `TextLit` in the fast case

Reply:
Ah yes, that makes sense. Thanks! 😄

Reply:
Alright, this case works now. Nice catch :smile: